### PR TITLE
Function for rotation in iterables

### LINKED ISSUE #16127: iterables could use rotations
Would a simple routine to return the rotations of a sequence be something useful in iterables?
```python
def rotations(s, dir=1):
    """Return a generator giving the items in s as list where
    each subsequent list has the items rotated to the left (default)
    or right (dir=-1) relative to the previous list.

    Examples
    ========

    >>> list(rotations([1,2,3]))
    [[1, 2, 3], [2, 3, 1], [3, 1, 2]]
    >>> list(rotations([1,2,3], -1))
    [[1, 2, 3], [3, 1, 2], [2, 3, 1]]
    """
    seq = list(s)
    for i in range(len(seq)):
        yield seq
        seq = rotate_left(seq, dir)
```

### PR DESCRIPTION
**Pull request by divyanshu132**
#### References to other Issues or PRs
Fixes #16127

#### Brief description of what is fixed or changed
Function yielding sequence rotations has been added to iterables.py of the utilities module.

#### Release Notes

*  utilities
   *  `rotations` function added to iterables.py

### CODE REVIEW

File: sympy/utilities/tests/test_iterables.py [hunk 1]
```diff
@@ -732,3 +732,14 @@ def test_ordered_partitions():
                 sum(1 for p in f(i, j, 1)) ==
                 sum(1 for p in f(i, j, 0)) ==
                 nT(i, j))
+
+
+def test_rotations():
+    assert list(rotations([10,-20,20])) == [[10, -20, 20], [-20, 20, 10], [20, 10, -20]]
```
Comment (smichr) [inline sympy/utilities/tests/test_iterables.py]: You can thin this out a bit: perhaps test `'ab'`, `range(3)`, and one of those with `dir=-1`.

**Comment by divyanshu132** [comment]
@smichr thanks, do this need more changes or it can be merged.

**Review by supreet11agrawal — APPROVED** [review]

**Review by supreet11agrawal — COMMENTED** [review]
I think in release notes, you should write `iterable.py` of `utilities` module. Will be more readable. Up to you

**Comment by asmeurer** [comment]
The `utilities` submodule is implied by the header. The bot adds each release notes entry under the appropriate header on the [release notes page](https://github.com/sympy/sympy/wiki/Release-Notes-for-1.4).

**Pull request merged**